1. Basics/task_7/part_3/task_3_on_apache_handler.php

```diff
@@ -0,0 +1,172 @@
+<?php
+/**
+ * Array contains data for response forming according its status code
+ */
+define('RESPONSE_DATA', [
+    '200' => ['statusMessage' => 'OK'],
+    '400' => ['statusMessage' => 'Bad Request'],
+    '404' => ['statusMessage' => 'Not Found'],
+]);
+
+processHttpRequest();
+
+// REQUEST PROCESS FUNCTION ///////////////////////////////////////////////////////////////////////
+/**
+ * Processes http request, partly forms the response. 
+ * Optionally outputs the results as HTML page.
+ * 
+ * @return void
+ */
+function processHttpRequest() : void {
+	$nums = testInputValue($_GET['nums'] ?? false);
+	$invalidRequest = checkRequestCorrectness($nums);
+	$isHTML = preg_match('#html#', $_SERVER['HTTP_ACCEPT'] ?? '');
+	
+	if($invalidRequest) {
+		http_response_code($invalidRequest['code']);
+		header('Error-Info: ' . $invalidRequest['info']);
+	}
+	
+	$responseBody = $invalidRequest ? ($invalidRequest['info'] ?? false) : array_sum(explode(',', $nums));	
+	
+	if($responseBody && !$isHTML) echo $responseBody;
+	
+	if($isHTML) outputHTML($responseBody);
+}
+
+/**
+ * Checks the request's correctness.
+ * If the request is correct return false.
+ * If the request is incorrect returns corresponding response code and some additional information.
+ *
+ * @param string|bool $nums
+ * @return bool|array
+ */
+function checkRequestCorrectness(string|bool $nums) : bool|array {
+	$methodGet = (($_SERVER['REQUEST_METHOD'] ?? null) === 'GET') ? true : false;				// 400
+	$correctUri = preg_match('#^/sum#', $_SERVER['REQUEST_URI'] ?? '');		// 404
```

Comment:
this way is not most optimal.
In any case you will do a lot of checks, even if not all of them are needed. For example, if method is invalid you anyway check uri and numbers...
no sense in this and you waste time and resources

Reply:
Caught it. Done.